### PR TITLE
feat: New Box Hackathon Feature on Townsquare

### LINKED ISSUE #6870: Feature  add Hackathon module on Townsquare Experiment
### User Story

As an operator of Gitcoin, we'd like more people to register for our hacakthons.

As a sponsor of Gitcoin, I'd like great devs to join my hackathons

As a developer, I'd like to know what is available for me to do on Gitcoin.

### Why Is this Needed

[comment]: # (Describe the problem and why this task is needed. Provide description of the current state, what you would like to happen, and what actually happen)
*Summary*:
We need more ways for people in townsquare to get started on hacakthons

_If we add this module, then our sponsors will be happy because more developers would register for Gitcoin hackathons because they'll know that a hackathon is going on or upcoming._

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Feature

### Current Behavior
No module to drive users into hackathons

### Expected Behavior
Display in right rail of townsquare

### Definition of Done
- Build the module on townsquare, right rail above referal link
• Show 2 current hackathons when a hackathon is happening
- Show 2 upcoming hackathons if there isn't one happening.
- “Join Hackathon” brings you to each hackathon onboarding • Show max 5 hackathons
- _View all_link at the bottom that links to Hackathon List page

### Data Requirements
Track clicks on # of registrants per hackathon

### Additional Information
Screen Shot 2020-06-15 at 2 49 15 PM

### PR DESCRIPTION
**Pull request by developerfred**
##### Description

New Box Hackathon Feature on Townsquare
Screen Shot 2020-06-16 at 18 46 17

##### Refers/Fixes
#6870

##### Testing

-  [movie screen here](https://share.getcloudapp.com/JruL1DzK)

Screen Shot 2020-06-22 at 10 38 02

### CODE REVIEW

**Comment by developerfred** [comment]
@PixelantDesign  Yes, the module is automatically hidden if there is no hackthon, I will add the click on the name.

**Comment by developerfred** [comment]
done @PixelantDesign

**Comment by developerfred** [comment]
@PixelantDesign yes!

**Comment by developerfred** [comment]
Thanks @thelostone-mc, I will change now.

**Comment by developerfred** [comment]
Screen Shot 2020-06-22 at 10 38 02

**Comment by danlipert** [comment]
please fill out the testing section

**Comment by developerfred** [comment]
@danlipert done!

File: app/townsquare/migrations/0020_auto_20200521_1020.py [hunk 1]
```diff
@@ -12,7 +12,7 @@ class Migration(migrations.Migration):
 
     operations = [
         migrations.AlterField(
-            model_name='pinnedpost',
+            model_name='PinnedPost',
```
Comment (danlipert) [inline app/townsquare/migrations/0020_auto_20200521_1020.py:15]: Whats up with this change?

Comment (developerfred) [reply app/townsquare/migrations/0020_auto_20200521_1020.py:15]: As I made the PinnedPost model, in lower case I was giving conflicts in my stay even deleting the entire project in Docker, if it is not necessary for production I can delete this file from the commit.

Comment (danlipert) [reply app/townsquare/migrations/0020_auto_20200521_1020.py:15]: that is very strange - yeah lets remove it and then if we need it we can do a separate PR so we can isolate the changes

**Comment by thelostone-mc** [comment]
@developerfred fixed up the indentation on html page
just looping you in so that in the following PRs -> we indent it the same 

Dope work btw ^_^

**Comment by owocki** [comment]
hey @developerfred up for a follow-up PR? https://github.com/gitcoinco/web/issues/7030

**Comment by octavioamu** [comment]
@developerfred seems is not working for svgs , hackathon logos are primary svg if not it use the png 
image

**Pull request merged**